### PR TITLE
[fastlane,fastlane_core] SWAG

### PR DESCRIPTION
**Pull request by hjanuschka**
as discussed on slack

for reference the initial gif:

### CODE REVIEW

**Comment by ohwutup** [comment]
This is so amazing! Will it run for every execution?

**Comment by hjanuschka** [comment]
@ohwutup for now yes - should we disable it by default enable it by `FastlaneCore::Env.truthy("FL_EASTEREGGS")` ?

**Comment by ohwutup** [comment]
Yeah! I think thats a good idea. Lets use something like `FL_ENABLE_LOGO_ANIMATION` or something like that, just so its a bit descriptive. 😄

**Comment by hjanuschka** [comment]
added `FL_ENABLE_LOGO_ANIMATION`  to enable the spinning logo.
its disabled if `!truthy? || !interactive? || is_ci?`

**Comment by KrauseFx** [comment]
> added `FL_ENABLE_LOGO_ANIMATION`  to enable the spinning logo.
> its disabled if `!truthy? || !interactive? || is_ci?`

Yes, that sounds great! Good thinking re interactive and ci. I'd also be in favour of having this as easter egg and disabled by default.

Once those changes are in, it's good to go from my side

**Comment by hjanuschka** [comment]
@KrauseFx  its disabled by default unless `FL_ENABLE_LOGO_ANIMATION` is set.

File: fastlane_core/lib/fastlane_core/swag.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
...xC2\xB4\xAE\x9C\x8E\xF9\xC1,\x88\xF1p\x10s\xB33T'\xB7=\x11\x0EE\xB5\x9A\x1F\x0F}!'\xC8\"%\xA4N\x8C\xCF\xAE\x95\xCC\xCEh\x97\x12rM,\x160\xD2%\x9A-\xAB\xEA\x11gQ\xB3zd\xF5Rr7\xF6\xE5\x03\xD0\n\xEB\x18\x9A\xC6C\xE2&\xB1\xEC\xDFj\xB5A\x8FyU\x97\"\x9EA\xB6\rwJ:o\x17\xB9=\\\x0EB3\xD2\xDAp~\x86\xC4\x00t\x1F\xE0\x90\x85S\xEA\n\x86\xCDC\xD5PO\x15\x00\v\x1A\xA9\x04`\x9D\x9C\xA1\xEE\xC7t}b\x95\x1D8\x1C\xC5R}\xDBH\xF9\xD5\x99\f\x83!!\x13;\x19A\x13\xEE\x9B\x96\xF9b\x10\x84\xE5\xD6\xD5\x18\x92\x96-\x06\t}\x10\xF6\x05\x056\xA4\x15\x1D(\x93\xD9\xAE\xF8`\x8E\x9C\xF1\x81\xECM\x00\x9B\xAB=\x1F-PUb\x93\x11&\xEDy\x92j\x90j\xDDk\x90\x00\xF8\x02[\xBF\xD6\x026\x8CX\x87M\xC3=\xE9\xA3\x9D%\x80.o\x1D8\xCCmG,w\xB4\xC0\x14\x87\xCB\x15~1\xFE\xCA\xF9'GS\fgyZ&Z\xC5\x80\x92\xFB&\xB1\xA5\xF6\xB3Z-\x1E!E\xD3\x9F\xD9\xC2\xA6y\x81;L\xD5\x1D\xB2\xD4\x88v\x9BA\x17\xC71\x9D5'\xD8?\x91\x9F\x9A\x81\xFB\xD1\x9B\xB4\x16v\xEC\xEE\xCB\xAE\x05\x85\xC7\xDC\e&\x13XA\x1C2q\xF4x\x9D\x8BQ$\x8F\xBC>\xD9\xC7\f\xC0\xB5\x15\xEC@\xFB\xA5\x1D\xD5\xC3\xF7+X\x7F\x00\xF7\xFBc\"\xA53D\xE40\xB0\xE1\xB4\xE8\b\x13\xEE\x9Fe\"\x10\xD2\xC5-\xCC\x91|\xBF\x0E7\xACW9\xA9\x18$\xE6\x8B\xA1\xDD$\xE7@\xC4\x8F\xA5\x83\x80\xFB\xD3\xD5\xB0\xF5\xC6\bye\x04J\x
```
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/swag.rb]: Oh, this shouldn't be here, right?

Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/swag.rb]: Same here

Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/swag.rb]: How about making it `should_be_shown?` instead?

Comment (hjanuschka) [inline fastlane_core/lib/fastlane_core/swag.rb]: @KrauseFx  i don't get it what do you mean?

**Comment by hjanuschka** [comment]
@KrauseFx  feedback addressed, renamed the method, and switched from if to unless - i hope in that case its fault free 👍

**Comment by radex** [comment]
@hjanuschka how about moving FRAMES [to the end of the file](https://shifteleven.com/articles/2009/02/09/useless-ruby-tricks-data-and-__end__/), after `__END__`? That would make it much easier to read the file.

**Comment by KrauseFx** [comment]
Wait, I thought this was merged already, so sad  😢

**Review by mgrebenets — APPROVED** [review]

**Pull request merged**